pkg/contracts/src/CVStrategy.sol

```diff
@@ -963,4 +874,95 @@ contract CVStrategy is BaseStrategy, IPointStrategy, ERC165 {
         _minThresholdPoints = minThresholdPoints_;
     }
+
+    function disputeProposal(uint256 proposalId, string calldata context, bytes calldata _extraData)
+        external
+        payable
+        nonReentrant
+    {
+        StrategyStruct.Proposal storage proposal = proposals[proposalId];
+
+        if (address(arbitrableConfig.arbitrator) == address(0)) {
+            revert ArbitratorCannotBeZero();
+        }
+        if (address(collateralVault) == address(0)) {
+            revert CollateralVaultCannotBeZero();
+        }
+        if (proposal.proposalId != proposalId) {
+            revert ProposalNotInList(proposalId);
+        }
+        if (proposal.proposalStatus != StrategyStruct.ProposalStatus.Active) {
+            revert ProposalNotActive(proposalId);
+        }
+        if (msg.value <= arbitrableConfig.collateralAmount) {
+            revert InsufficientCollateral(msg.value, arbitrableConfig.collateralAmount);
+        }
+
+        uint256 arbitrationFee = msg.value - arbitrableConfig.collateralAmount;
+
+        collateralVault.depositCollateral{value: arbitrableConfig.collateralAmount}(proposalId, msg.sender);
+
+        uint256 disputeId = arbitrableConfig.arbitrator.createDispute{value: arbitrationFee}(RULING_OPTIONS, _extraData);
+
+        proposal.proposalStatus = StrategyStruct.ProposalStatus.Disputed;
+        proposal.disputeId = disputeId;
+        proposal.disputeTimestamp = block.timestamp;
+        proposal.challenger = msg.sender;
+        disputeIdToProposalId[disputeId] = proposalId;
+        emit ProposalDisputed(proposalId, disputeId, msg.sender, arbitrationFee, context);
+    }
+
+    function rule(uint256 _disputeID, uint256 _ruling) external override {
+        uint256 proposalId = disputeIdToProposalId[_disputeID];
+        StrategyStruct.Proposal storage proposal = proposals[proposalId];
+
+        if (proposalId == 0) {
+            revert ProposalNotInList(proposalId);
+        }
+        if (proposal.proposalStatus != StrategyStruct.ProposalStatus.Disputed) {
+            revert ProposalNotDisputed(proposalId);
+        }
+
+        bool isTimeOut = block.timestamp > proposal.disputeTimestamp + arbitrableConfig.defaultRulingTimeout;
+
+        if (!isTimeOut && msg.sender != address(arbitrableConfig.arbitrator)) {
+            revert OnlyArbitrator();
+        }
+
+        if (isTimeOut || _ruling == 0) {
+            if (arbitrableConfig.defaultRuling == 1) {
+                proposal.proposalStatus = StrategyStruct.ProposalStatus.Active;
+            }
+            if (arbitrableConfig.defaultRuling == 2) {
+                proposal.proposalStatus = StrategyStruct.ProposalStatus.Blocked;
+            }
+            collateralVault.withdrawCollateral(proposalId, proposal.challenger, arbitrableConfig.collateralAmount);
+            collateralVault.withdrawCollateral(proposalId, proposal.submitter, arbitrableConfig.collateralAmount);
```

Comment:
The submitter proposal should go back to collateral vault
![image](https://github.com/user-attachments/assets/dcd9d9a1-5759-43a6-a7c6-136fe8c85536)

Reply:
And will be restored when proposal cancelled or executed

Reply:
Yes, we will need a withdrawFor function in order to withdraw in name of other like ` function withdrawCollateral(uint256 proposalId, address user, address to, uint256 amount)`